pkg/discovery/repository/kube_cluster.go

```diff
@@ -192,6 +195,67 @@ func (summary *ClusterSummary) computePodToServiceMap() {
 	}
 }
 
+func (summary *ClusterSummary) computeStaticPodToDaemonMap() {
+	glog.V(2).Info("creating mapping for static pods modeled as daemons.")
+	// static pods are only exposed through the API via mirror pods
+	// https://kubernetes.io/docs/reference/glossary/?all=true#term-mirror-pod
+	mirrorPods := util.GetMirrorPods(summary.Pods)
+	// common name prefix is our assumed conventions for showing intent of creating
+	// daemon sets with static pods
+	prefixToNodeNames := util.GetMirrorPodPrefixToNodeNames(mirrorPods)
+	nodePoolToNodeNames := util.MapNodePoolToNodeNames(summary.Nodes)
+	anyPools := len(nodePoolToNodeNames) != 0
+	if anyPools {
+		glog.V(3).Info("Conducting static pod daemon check with node pools.")
+	} else {
+		glog.V(3).Info("Conducting static pod daemon check with nodes.")
+	}
+
+	staticPodToDaemonMap := make(map[string]bool)
+	prefixToDaemon := make(map[string]bool)
+	daemonCount := 0
+	for _, pod := range mirrorPods {
+		prefix, ok := util.GetMirrorPodPrefix(pod)
+		nodeNames := prefixToNodeNames[prefix]
+		// check whether the prefix has been processed before to prevent unnecessary work
+		if value, exists := prefixToDaemon[prefix]; ok && exists {
+			if value {
+				daemonCount++
+			}
+			staticPodToDaemonMap[string(pod.UID)] = value
+			continue
+		}
+
+		var daemon = true
+		if anyPools {
+			for _, nodePool := range nodePoolToNodeNames {
+				if intersection := nodePool.Intersection(nodeNames); intersection.Len() == 0 {
+					daemon = false
+					break
+				}
+			}
+		} else {
+			for _, node := range summary.Nodes {
+				if !nodeNames.Has(node.GetName()) {
+					daemon = false
+					break
+				}
+			}
+		}
```

Comment:
Again, this part of the logic could be simplified if we treat the entire cluster as a single node pool, it should not be treated any different than other node pools.

Also, I believe the logic should be:

* Initialize `daemon` to false
* As long as one node pool matches the nodeSet that corresponds to a prefix, then set `daemon` to true and break